crates/ide_assists/src/handlers/extract_function.rs

```diff
@@ -476,11 +472,13 @@ impl FunctionBody {
     }
 
     fn from_range(parent: ast::StmtList, selected: TextRange) -> FunctionBody {
-        let mut text_range = parent
-            .statements()
-            .map(|stmt| stmt.syntax().text_range())
+        let full_body = parent.syntax().children_with_tokens();
+
+        let mut text_range = full_body
+            .map(|stmt| stmt.text_range())
```

Comment:
Let's do a `filter` in front of the map with `filter(|it| matches!(it.kind(), STMT | COMMENT))`, that way we don't iterate the braces of the `StmtList`, only the things we are interested in

Reply:
I looked around on this for a while but I can't seem to find any `STMT`. I did come up with this alternative but I'm not quite satisfied with it. Do you have any suggestions here?

[`51c50dd` (#11322)](https://github.com/rust-analyzer/rust-analyzer/pull/11322/commits/51c50dd5acd34fac006406a6fc24f51a624e208e)

Reply:
Ah that is my bad, I forgot ast::Stmt is an enum, therefor no such SyntaxKind exists.

`filter(|it| ast::Stmt::can_cast(it.kind()) || it.kind() == COMMENT)`

should work

Reply:
🙏 Thanks for clarifying that, it works like a charm! I've made the changes you requested